From 5a0eeb72c3cde7dcb8096967561a88a678ad9aec Mon Sep 17 00:00:00 2001
From: Ashish Kumar Dhanotiya <adhanoti@codeaurora.org>
Date: Fri, 8 Dec 2017 14:50:46 +0530
Subject: wlan: Return an error code in case of invalid seq_len while adding
 key

There is no return statement in case of invalid sequence length in
__wlan_hdd_cfg80211_add_key api.

Return an error code in case of invalid seq_len.

Change-Id: Ib240f6cf91d9d563dd87148c15e06ae8b48919d5
CRs-Fixed: 2153553
---
 CORE/HDD/src/wlan_hdd_cfg80211.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CORE/HDD/src/wlan_hdd_cfg80211.c b/CORE/HDD/src/wlan_hdd_cfg80211.c
index eabee7b..a0505a9 100644
--- a/CORE/HDD/src/wlan_hdd_cfg80211.c
+++ b/CORE/HDD/src/wlan_hdd_cfg80211.c
@@ -11770,6 +11770,8 @@ static int __wlan_hdd_cfg80211_add_key( struct wiphy *wiphy,
     {
         hddLog(VOS_TRACE_LEVEL_ERROR,"%s: Invalid seq length %d", __func__,
                 params->seq_len);
+
+        return -EINVAL;
     }
 
     hddLog(VOS_TRACE_LEVEL_INFO,
-- 
cgit v1.1

